### DISCUSSION TITLE
Support for https imports

### DISCUSSION THREAD

**Discussion by brc-dd** [jsr-io/jsr#674] [General]
Currently, JSR doesn't allow https imports. This has some issues if someone is writing modules specifically for deno:

- One cannot depend on any third-party module that's not published to JSR or NPM. For example, the modules from deno.land/x/
- The globals in modules imported using `npm:` are different than deno ones. So, if you're using some npm module that patches them, it simply won't work for calls from deno. The only way to handle this was using esm.sh, which we can't use with JSR.

Proposed solution:

- Allow importing https modules in packages. These won't work in runtimes other than deno and browsers. So leave it up to the module author to specify compatibility in web settings.
- If security is a concern, maybe safelist only some domains like deno.land/x/ and esm.sh

PS: this is not about importing JSR modules using HTTP URLs, it's about using HTTP imports in published files.

**Comment by xzdwq** [comment]
Hi!
I would like to add my own idea.
By publishing packages in JSR, we get the opportunity to immediately work with them as with ECMAScript modules, but inspired by the http imports of modules into Deno and slightly reconsidering the approach, I would like to suggest the following:

Be able to receive a file from JSR directly over http, for example:
```typescript
import * as X from 'https://raw.jsr.io/@xzdwq/test/0.1.0/mod.ts';
```
Specifying the response header `Content-type: application/javascript;`

This is convenient when you need to use a specific script or function from a file that is implemented purely without other dependencies.

For security reasons, you can set the `raw` parameter on the client to receive the file as text, for example:
```typescript
import * as X from 'https://raw.jsr.io/@xzdwq/test/0.1.0/mod.ts?raw';
```
Specifying the response header `Content-type: text/plain;`

It would also be convenient to use it in Bun due to the Virtual modules mechanism: https://bun.sh/docs/runtime/plugins#virtual-modules

The main motivation is variations in the use of modules and functions, their independent support, like with a micro-frontens.

**Reply by brc-dd** [reply]
This already works. You can directly import files from https://jsr.io. Type checking won't work, although intellisense works fine:

```ts
import { assert } from 'https://jsr.io/@std/assert/1.0.2/assert.ts'

assert(false)
```
